### ISSUE TITLE
Lab 06: Test case 8 

### ISSUE THREAD

**Issue by tranquilityyy** [nus-cs2030/2223-s1#259]
I'm currently facing a situation where after my server rest has been implemented, somehow my customer can no longer tell that a queue is full (second screenshot below shows how the server is reflected to have an empty queue (before adding customer) despite it happening right after the previous event where the queue is added.)

This problem does not occur for other situations where there is no rest time, but I cannot seem to think of how my rest time implementation could possibly cause this error.

Really last minute before the deadline but would appreciate if anyone has any input!

![image](https://user-images.githubusercontent.com/114040935/199651151-af426ac0-f2fb-4eac-b5bd-c9f3f591adc5.png)

![image](https://user-images.githubusercontent.com/114040935/199651298-a56ee55b-1c7e-4a3b-be32-e0d71e4a10a2.png)

**Comment by dohaduong** [comment]
It seems like when your customer 6 arrives, instead of checking with the updated server 1 (updated with customer 5 in queue), it checks with the un-updated server 1 (the resting one before 5 comes).
I think you could check if the server and its waiting queue are updated correctly when your server is resting.

**Comment by glendon29** [comment]
most likely you have updated the queue wrongly. Ensure that the customer always retrieve the last updated queue when it arrives

**Comment by cherylchan02** [comment]
maybe you should check how you update the servers? customer 6 is not checking the most updated server. could you share how you updated the servers? i think the because of the rest time, the updating of the servers may not doing what you expect it to do.

**Comment by giorgiocana** [comment]
Hi, if you are sure that your rest times are correct, you should check 2 things:

1. How is your nextEvent method implemented when your customer arrives?  Remember that if the server is resting, a costumer can still join its queue. (I had the same problem and fixed it by modifying my arrive class);
2. Check that your server queues are updating correctly. Try printing out all the queus before and after updating them each time, that also helped me solve lots of issues with my code.

Hope this helps!

**Comment by tranquilityyy** [comment]
Thanks to all for the suggestions! Finally managed to got it to work. 

Solution: 
I have checked in detail how the waiting queue has been updated with each of my event, and for my implementation I realised it's not because my server did not update after the previous person comes, but instead it's because I've emptied the queue in advance again (e.g. if Customer 5 in this example is supposed to be served at 3.9 if the server does not rest, but instead at 4.315 if server rests --> instead of only emptying the queue at 4.315, my queue was emptied at 3.9 and directly passed to the Arrive event at 3.910, hence confusing the server).

It was ultimately helpful to trace the state of the queue between every single event, be it hidden/not (given that I implemented hidden events for myself).
